### PR TITLE
LX539, MemoryTypeIndices not validated at BindImageMemory/BindBufferMemoryTime

### PR DESCRIPTION
**Pull request by mark-lunarg**

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Doc update to get added shortly.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -5138,6 +5138,20 @@ VKAPI_ATTR void VKAPI_CALL DestroyImage(VkDevice device, VkImage image, const Vk
...
 }
 
+static bool validate_memory_types(const layer_data *dev_data, const DEVICE_MEM_INFO *mem_info, const uint32_t memory_type_bits,
+                                  const char *funcName) {
+    bool skip_call = false;
+    if (((1 << mem_info->allocInfo.memoryTypeIndex) & memory_type_bits) == 0) {
+        skip_call = log_msg(dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_DEVICE_MEMORY_EXT,
+                            reinterpret_cast<const uint64_t &>(mem_info->mem), __LINE__, MEMTRACK_INVALID_MEM_TYPE, "MT",
+                            "%s(): MemoryRequirements->memoryTypeBits (%d) for this object type are not compatible with the memory "
+                            "type (%d) of this memory object 0x%" PRIx64 ".",
```
Comment (tobine) [inline layers/core_validation.cpp]: I recommend printing both of the bitfields in hex

**Comment by tobine** [comment]
Comment on print, but otherwise LGTM 👍

**Comment by mark-lunarg** [comment]
Updated prints, pushed to Master.

commit 4261f1e2c91c90ce040ab36a13c9d92f6e988f10
Author: Mark Lobodzinski <mark@lunarg.com>
Date:   Tue Jun 14 12:11:29 2016 -0600

    layers: LX539, Validate mem types

    Bind[Image|Buffer]Memory did not validate that the memoryPropertyType
    matched the memory_requirements memory type flags for the buffer
    or image.

**Pull request closed without merging**